generator/discovery/openstack/vm_images_discovery.py

```diff
@@ -0,0 +1,755 @@
+        Return one credential for each VM image provided by openstack cloud.
+
+        @return: list of VM images
+        @rtype: list[JsonLdObject]
+        """
+        return [
+            JsonLdObject(self._convert_to_gx_image(image), gx_id=image.id)
+            for image in self.conn.list_images()
+            if image.visibility == "public"
+        ]
+
+    def _convert_to_gx_image(self, os_image: OS_Image) -> GX_Image:
+        """
+        Convert Openstack image to a Gaia-X virtual machine image.
+        @param os_image: Openstack image properties
+        @type os_image: OS_Image
+        @return: Gaia-X virtual machine image
+        @rtype GX_Image
+        """
+
+        # Initialize Gaia-X Image
+        gx_image = GX_Image(
+            copyrightOwnedBy="TBA",
+            license="TBA",
+            resourcePolicy=const.DEFAULT_RESOURCE_POLICY,
+        )
+
+        # Discover optional attributes
+        gx_image.vmImageDiskFormat = self._get_disk_format(os_image)
+        gx_image.secureBoot = self._get_secure_boot(os_image)
+        gx_image.firmwareType = self._get_firmeware_type(os_image)
+        gx_image.watchDogAction = self._get_watchdog_action(os_image)
+        gx_image.vPMU = self._get_vmpu(os_image)
+        gx_image.cpuReq = self._get_cpu_req(os_image)
+        gx_image.multiQueues = self._get_multiqueue_enabled(os_image)
+        gx_image.checksum = self._get_checksum(os_image)
+        gx_image.hwRngTypeOfImage = self._get_rng_model(os_image)
+        gx_image.videoRamSize = self._get_video_ram_size(os_image)
+        gx_image.file_size = self._get_file_size(os_image)
+        gx_image.updateStrategy = self._get_update_strategy(os_image)
+        gx_image.description = self._get_description(os_image)
+        gx_image.name = self._get_name(os_image)
+        gx_image.ramReq = self._get_min_ram_req(os_image)
+        gx_image.rootDiskReq = self._get_min_disk_req(os_image)
+        gx_image.operatingSystem = self._get_operation_system(os_image)
+        gx_image.buildDate = self._get_build_date(os_image)
+        gx_image.licenseIncluded = self._get_license_included(os_image)
+        gx_image.patchLevel = self._get_patch_level(os_image)
+        gx_image.version = self._get_version(os_image)
+        gx_image.maintenance = self._get_maintenance(os_image)
+        gx_image.signature = self._get_signature(os_image)
+        gx_image.hypervisorType = self._get_hypervisor_type(os_image)
+
+        # Discover mandatory attribute
+        self._add_license(os_image, gx_image)
+        self._add_copyright_owner(os_image, gx_image)
+        self._add_resource_policy(os_image, gx_image)
+
+        return gx_image
+
+    @staticmethod
+    def _get_disk_format(os_image: OS_Image) -> VMDiskType:
+        if os_image.disk_format:
+            return VMDiskType(
+                DISK_LOOKUP.get(os_image.disk_format.lower(), VMDiskType.RAW.text)
+            )
+        else:
+            return VMDiskType("RAW")
```

Comment:
even better (again, **provided that disk_format can be None**)

```suggestion
        if os_image.disk_format is None:
            return VMDiskType(VMDiskType.RAW.text)
        return VMDiskType(
            DISK_LOOKUP.get(os_image.disk_format.lower(), VMDiskType.RAW.text)
        )
```